sympy/solvers/tests/test_solveset.py

```diff
@@ -1749,3 +1730,187 @@ def test_issue_14454():
+    assert solveset(2**x + 4**x + 8**x, x) == Union(union1, union2)
+
+    eq = 4**(x + 1) + 4**(x + 2) + 4**(x - 1) - 3**(x + 2) - 3**(x + 3)
+    res = solveset(eq, x)
+    num = 2*n*I*pi - 4*log(2) + 2*log(3)
+    den = -2*log(2) + log(3)
+    ans = imageset(Lambda(n, num/den), S.Integers)
+    assert res == ans
+
+
+def test_expo_conditionset():
+    from sympy.abc import x, y
+
+    f1 = (exp(x) + 1)**x - 2
+    f2 = (x + 2)**y*x - 3
+    f3 = 2**x - exp(x) - 3
+    f4 = log(x) - exp(x)
+
+    assert solveset(f1, x, S.Reals) == ConditionSet(
+        x, Eq((exp(x) + 1)**x - 2, 0), S.Reals)
+    assert solveset(f2, x, S.Reals) == ConditionSet(
+        x, Eq(x*(x + 2)**y - 3, 0), S.Reals)
+    assert solveset(f3, x, S.Reals) == ConditionSet(
+        x, Eq(2**x - exp(x) - 3, 0), S.Reals)
+    assert solveset(f4, x, S.Reals) == ConditionSet(
+        x, Eq(-exp(x) + log(x), 0), S.Reals)
+
+
+def test_exponential_symbols():
+    x, y, z = symbols('x y z', positive=True)
+    from sympy import simplify
+
+    assert solveset(z**x - y, x, S.Reals) == Intersection(
+        S.Reals, FiniteSet(log(y)/log(z)))
+
+    w = symbols('w')
+    f1 = 2*x**w - 4*y**w
+    f2 = (x/y)**w - 2
+    ans1 = solveset(f1, w, S.Reals)
+    ans2 = solveset(f2, w, S.Reals)
+    assert ans1 == simplify(ans2)
+
+    assert solveset(x**x, x, S.Reals) == S.EmptySet
+    assert solveset(x**y - 1, y, S.Reals) == FiniteSet(0)
+    assert solveset(exp(x/y)*exp(-z/y) - 2, y, S.Reals) == FiniteSet(
+        (x - z)/log(2)) - FiniteSet(0)
+
+
+@XFAIL
+def test_issue_10864():
+    assert solveset(x**(y*z) - x, x, S.Reals) == FiniteSet(1)
+
+
+@XFAIL
+def test_solve_only_exp_2():
+    assert solveset_real(sqrt(exp(x)) + sqrt(exp(-x)) - 4, x) == \
+        FiniteSet(2*log(-sqrt(3) + 2), 2*log(sqrt(3) + 2))
+
+
+def test_is_exponential():
+    x, y, z = symbols('x y z')
+
+    assert _is_exponential(3**x - 2, x) is True
+    assert _is_exponential(5**x - 7**(2 - x), x) is True
+    assert _is_exponential(sin(2**x) - 4*x, x) is False
+    assert _is_exponential(x**y - z, y) is True
+    assert _is_exponential(2**x + 4**x - 1, x) is True
+    assert _is_exponential(x**(y*z) - x, x) is False
+
+
+def test_solve_expo():
+    assert _solve_expo(3**(2*x) - 2**(x + 3), x) == \
+        2*x*log(3) - (x + 3)*log(2)
+    y = symbols('y')
+    assert _solve_expo(2**y + 4**y, y) == \
+        log(2**y) - log(-4**y)
+    assert _solve_expo(2**x + 3**x + 5**x, x) is None
+
+# end of exponential tests
+
+
+# logarithmic tests
+@XFAIL
+def test_uselogcombine_1():
+    assert solveset_real(log(x - 3) + log(x + 3), x) == FiniteSet(
+            sqrt(10))
+    assert solveset_real(log(x + 1) - log(2*x - 1), x) == FiniteSet(2)
+    assert solveset_real(log(x + 3) + log(1 + 3/x) - 3) == FiniteSet(
+        -3 + sqrt(-12 + exp(3))*exp(S(3)/2)/2 + exp(3)/2,
+        -sqrt(-12 + exp(3))*exp(S(3)/2)/2 - 3 + exp(3)/2)
+
+
+@XFAIL
+def test_uselogcombine_2():
```

Comment:
Instead of 1 or 2 as function name suffix, write the difference between them as function name.

Reply:
I don't see a difference here, these tests were already present I just moved them here.
I would rather include all in one.